backend/dataall/api/Objects/ShareObject/resolvers.py

```diff
@@ -231,16 +231,6 @@ def resolve_user_role(context: Context, source: models.ShareObject, **kwargs):
             or dataset.SamlAdminGroupName in context.groups
         ):
             return ShareObjectPermission.Requesters.value
-        if (
-            dataset and dataset.stewards in context.groups
-            and (
-                source.owner == context.username
-                or source.principalId in context.groups
-                or dataset.owner == context.username
-                or dataset.SamlAdminGroupName in context.groups
-            )
-        ):
-            return ShareObjectPermission.ApproversAndRequesters.value
         else:
```

Comment:
If I look at the logic of this, we are classifying users into Approvers, Requesters and NoPermissions.
Approvers should be = dataset owners and stewards
Requesters = share owners
NoPermissions = none of the above

Following this definitions, we should have something like:
```
       if dataset and (
                dataset.stewards in context.groups 
                or dataset.SamlAdminGroupName in context.groups
                or dataset.owner == context.username
       ):
            return ShareObjectPermission.Approvers.value
        if (
            source.owner == context.username
            or source.principalId in context.groups
        ):
            return ShareObjectPermission.Requesters.value
```

We probably did not spot this issue because we were not using additional stewards (by default if no stewards are selected it defaults to the dataset.SamlAdminGroupName)

Reply:
We didn't spot the issue because we have two checks that are almost identical. There is no way that the second check will be true, but first not.
The first check that returns Approvers.value: 
```
dataset and dataset.stewards in context.groups 
```
Second check that should retrun broken value ApproversAndRequesters:
```
dataset and dataset.stewards in context.groups and (more checks....)
```
So basically, this is a dead code) 

If you are still sure that the `Approvers` logic should be changed as you suggested, please let know I'll do it :)

Reply:
Let's see it with an example, because I am not convinced hahahah
Dataset: DatasetA1
- owner: userA
- SamlAdminGroupName: groupA
- Stewards: None --> it defaults to groupA
ShareRequest: ShareB
- owner: userB
- principalId: groupB

groupA should always be classified as Approvers and groupB as Requesters



Dataset: DatasetA2
- owner: userA
- SamlAdminGroupName: groupA
- Stewards: groupA2
ShareRequest: ShareB
- owner: userB
- principalId: groupB

Reply:
Got it! I thought that we were discussing about why the check is not hit, but you are suggesting to change the logic in the check. I'll gladly do that :)